include/boost/wave/util/cpp_macromap.hpp

```diff
@@ -1004,13 +1023,56 @@ bool adjacent_stringize = false;
                     impl::replace_ellipsis(expanded_args, i, expanded, pos);
                 }
                 else
+
+#if BOOST_WAVE_SUPPORT_VA_OPT != 0
+                if (is_va_opt) {
+                    position_type const &pos = (*cit).get_position();
+
+                    BOOST_ASSERT(boost::wave::need_va_opt(ctx.get_language()));
+
+                    // ensure all variadic arguments to be expanded
+                    for (typename vector<ContainerT>::size_type arg = i;
+                         arg < expanded_args.size(); ++arg)
+                    {
+                        expand_argument(arg, arguments, expanded_args,
+                            expand_operator_defined, has_expanded_args);
+                    }
+
+                    // locate the __VA_OPT__ arguments
+                    typename macro_definition_type::const_definition_iterator_t cstart = cit;
+                    if (!impl::find_va_opt_params(cit, cend)) {
+                        BOOST_WAVE_THROW_CTX(ctx, preprocess_exception,
+                            improperly_terminated_macro, "missing ')' in __VA_OPT__",
+                            pos);
+                    }
+                    // cstart still points to __VA_OPT__; cit now points to the last rparen
+
+                    if ((i == arguments.size()) ||             // no variadic argument
+                        impl::is_blank_only(arguments[i])) {   // no visible tokens
+                        // no args; insert placemarker
+                        *std::back_inserter(expanded) =
+                            typename ContainerT::value_type(T_PLACEMARKER, "\xA7", pos);
+                    } else {
+                        ++cstart; ++cstart;
```

Comment:
I think we should make sure that the replacement list does not contain a `__VA_OPT__` itself (P0306r4 states:
> The identifier __VA_OPT__ shall always occur as part of the token sequence __VA_OPT__(content), where content is an arbitrary sequence of preprocessing-tokens other than __VA_OPT__.

Reply:
I will add a test case and whatever code is required to properly check